ddp-workspace/projects/ddp-dsm-ui/src/app/field-datepicker/field-datepicker.component.ts

```diff
@@ -83,43 +90,56 @@ export class FieldDatepickerComponent implements OnInit, OnChanges {
   }
 
   public check(): void {
-    this.colorDuringPatch = true;
-    if (this._dateString != null) {
-      if (this._dateString === '') {
-        this.emitDate('');
-      } else {
-        if (this._dateString !== 'N/A' && this._dateString !== 'Not Found') {
-          const tmp = Utils.parseDate(this._dateString, this.dateFormat, this.allowUnknownDay);
-          // console.log(tmp);
-          if (tmp instanceof Date) {
-            this.datePicker = tmp;
-            if (this.datePicker != null) {
-              this.emitDate(Utils.getDateFormatted(this.datePicker, Utils.DATE_STRING));
-              this.colorDuringPatch = false;
-              this.error = null;
-            } else {
-              this.error = this.dateFormat;
-            }
-          } else if (tmp === Utils.DATE_PARTIAL) {
+    if (this._dateString === '') {
+      this.emitDate('');
+    } else {
+      if (this._dateString !== 'N/A' && this._dateString !== 'Not Found') {
+        const tmp = Utils.parseDate(this._dateString, this.dateFormat, this.allowUnknownDay);
+        // console.log(tmp);
+        if (tmp instanceof Date) {
+          this.hasDateChanged = (this.datePicker !== tmp);
+          this.datePicker = tmp;
+          if (this.datePicker != null) {
+            this.emitDate(Utils.getDateFormatted(this.datePicker, Utils.DATE_STRING));
             this.error = null;
-            this.emitDate(Utils.getPartialDateFormatted(this._dateString, this.dateFormat));
-            this.colorDuringPatch = false;
-          } else if (tmp == null) {
+          } else {
             this.error = this.dateFormat;
           }
+        } else if (tmp === Utils.DATE_PARTIAL) {
+          this.error = null;
+          this.emitDate(Utils.getPartialDateFormatted(this._dateString, this.dateFormat));
+          this.hasDateChanged = false;
+        } else if (tmp == null) {
+          this.error = this.dateFormat;
         }
       }
-    } else {
-      this.colorDuringPatch = false;
-      this.error = null;
     }
   }
 
+  private async saveDate(e: Event): Promise<void>{
+    this.saveButtonText = "Saving..."
+    try{
+      this.isSaved = await this.dateSaved();
+      this.saveButtonText = this.isSaved ? "Success!": "Failed";
+      this.hasDateChanged = false;
+    }
+    catch(e){
+        this.saveButtonText = "Failed";
+        console.log(e)
+    }
+    setTimeout(() => {
+      this.saveButtonText = "Save Date"
+    },5000);
```

Comment:
What if the request took longer than 5 seconds? Should we still show "Save Date", even if the request is still pending?

Reply:
I will put it inside a finally so the request completes before the timeout begins